src/uu/top/src/header.rs

```diff
@@ -0,0 +1,362 @@
+        + steal_time
+        + guest
+        + guest_nice;
+
+    format!(
+        "%Cpu(s):  {:.1} us, {:.1} sy, {:.1} ni, {:.1} id, {:.1} wa, {:.1} hi, {:.1} si, {:.1} st",
+        user / total * 100.0,
+        system / total * 100.0,
+        nice / total * 100.0,
+        idle / total * 100.0,
+        io_wait / total * 100.0,
+        hardware_interrupt / total * 100.0,
+        software_interrupt / total * 100.0,
+        steal_time / total * 100.0,
+    )
+}
+
+#[cfg(target_os = "windows")]
+fn cpu() -> String {
+    use libc::malloc;
+    use windows::Wdk::System::SystemInformation::NtQuerySystemInformation;
+
+    #[repr(C)]
+    #[derive(Debug)]
+    struct SystemProcessorPerformanceInformation {
+        idle_time: i64,       // LARGE_INTEGER
+        kernel_time: i64,     // LARGE_INTEGER
+        user_time: i64,       // LARGE_INTEGER
+        dpc_time: i64,        // LARGE_INTEGER
+        interrupt_time: i64,  // LARGE_INTEGER
+        interrupt_count: u32, // ULONG
+    }
+
+    let n_cpu = sysinfo().read().unwrap().cpus().len();
+    let mut cpu_load = SystemProcessorPerformanceInformation {
+        idle_time: 0,
+        kernel_time: 0,
+        user_time: 0,
+        dpc_time: 0,
+        interrupt_time: 0,
+        interrupt_count: 0,
+    };
+    unsafe {
+        let len = n_cpu * size_of::<SystemProcessorPerformanceInformation>();
+        let data = malloc(len);
+        let _ = NtQuerySystemInformation(
+            windows::Wdk::System::SystemInformation::SystemProcessorPerformanceInformation,
+            data,
+            (n_cpu * size_of::<SystemProcessorPerformanceInformation>()) as u32,
+            std::ptr::null_mut(),
+        );
+        for i in 0..n_cpu {
+            let cpu = data.add(i * size_of::<SystemProcessorPerformanceInformation>())
+                as *const SystemProcessorPerformanceInformation;
+            let cpu = cpu.as_ref().unwrap();
+            cpu_load.idle_time += cpu.idle_time;
+            cpu_load.kernel_time += cpu.kernel_time;
+            cpu_load.user_time += cpu.user_time;
+            cpu_load.dpc_time += cpu.dpc_time;
+            cpu_load.interrupt_time += cpu.interrupt_time;
+            cpu_load.interrupt_count += cpu.interrupt_count;
+        }
+    }
+    let total = cpu_load.idle_time
+        + cpu_load.kernel_time
+        + cpu_load.user_time
+        + cpu_load.dpc_time
+        + cpu_load.interrupt_time;
+    format!(
+        "%Cpu(s):  {:.1} us,  {:.1} sy,  {:.1} id,  {:.1} hi,  {:.1} si",
+        cpu_load.user_time as f64 / total as f64 * 100.0,
+        cpu_load.kernel_time as f64 / total as f64 * 100.0,
+        cpu_load.idle_time as f64 / total as f64 * 100.0,
+        cpu_load.interrupt_time as f64 / total as f64 * 100.0,
+        cpu_load.dpc_time as f64 / total as f64 * 100.0,
+    )
+}
+
+//TODO: Implement for macos
+#[cfg(target_os = "macos")]
+fn cpu() -> String {
+    todo()
+}
+
+fn memory(arg: &ArgMatches) -> String {
```

Comment:
instead of passing ArgMatches, you could just pass arg.get_one::<String>("scale-summary-mem")

would be easier to understand + easier to test this function :)